### ISSUE TITLE
Memory leak in stub mode

### ISSUE THREAD

**Issue by bestwnh** [Moya/Moya#1000]
The demo project can be reproduce the issue.
Just replace the 
```swift
downloadRepositories("ashfurrow")
```
with 
```swift
MoyaProvider<GitHub>(stubClosure:MoyaProvider.immediatelyStub).request(GitHub.userRepositories("test")) { (result) in
    print(result)
}
```
Then it show up with memory leak
<img width="1222" alt="screen shot 2017-03-06 at 16 05 10" src="https://cloud.githubusercontent.com/assets/1201345/23601575/2d455bba-0287-11e7-87be-9ab1283931f9.png">
<img width="1207" alt="screen shot 2017-03-06 at 16 09 42" src="https://cloud.githubusercontent.com/assets/1201345/23601615/5b5869ca-0287-11e7-8af8-0cf78b41af05.png">

But something wired:
This leak only happen with stub mode, work fine on real request.
```swift
✅ MoyaProvider.neverStub
❌ MoyaProvider.immediatelyStub
❌ MoyaProvider.delayStub()
```
But if I use the `GitHubProvider` in `GitHubApi.swift` to call in stub mode(Just change the stub mode in `GitHubApi.swift`), it won't leak for all situation.

**Comment by bestwnh** [comment]
Is there anyone can help with this? Which file I should look into?

**Comment by scottrhoyt** [comment]
@bestwnh. We'd love to have your support in taking a look at this if you can. I would start looking in these locations:

Where the code branches from a normal request to a stub request:
https://github.com/Moya/Moya/blob/master/Sources/Moya/MoyaProvider%2BInternal.swift#L71

The function that is called to stub a request:
https://github.com/Moya/Moya/blob/master/Sources/Moya/MoyaProvider.swift#L92

**Comment by bestwnh** [comment]
@scottrhoyt I created a pull request to fix this issue. Your tips really save my time.

**Comment by BasThomas** [comment]
Solved in #1002
